src/midi/midi_lasynth_model.cpp

```diff
@@ -49,6 +50,23 @@ LASynthModel::LASynthModel(const std::string &rom_name,
 	assert(ctrl_full || (ctrl_a && ctrl_b));
 }
 
+// Checks the directory for the ROM name in a case insensitive manner.
+static std::optional<std_fs::path> GetROMPath(const std_fs::path& dir, const std::string& filename)
+{
+	std::string filename_lower = filename;
+	lowcase(filename_lower);
+	try {
+		for (const auto& entry : std_fs::directory_iterator(dir)) {
```

Comment:
The try will bail out on the first  problem (maybe a broken symlink), and we won't be able to drill in further to get the files. 
Suggest using the error-checking versions of all the filesystem calls here to continue past problems (and avoid exception handling entirely).;

Reply:
Also, the function is very generic and useful for other file lookups (perhaps SF2 and CDROM image files). 

Suggest adding it to fs_utils

Reply:
I just moved the function to fs_utils as you suggested.  I switched to the error code version of `directory_iterator` which I believe is the only function that can throw an exception.  Let me know if that looks right to you.  I don't quite know all the ins and outs of this C++ 17 filesystem API.